krkn/resiliency/resiliency.py

```diff
@@ -0,0 +1,409 @@
+"""Resiliency evaluation orchestrator for Krkn chaos runs.
+
+This module provides the `Resiliency` class which loads the canonical
+`alerts.yaml`, executes every SLO expression against Prometheus in the
+chaos-test time window, determines pass/fail status and calculates an
+overall resiliency score using the generic weighted model implemented
+in `krkn.resiliency.score`.
+"""
+
+from __future__ import annotations
+
+import base64
+import datetime
+import logging
+import os
+from typing import Dict, List, Any, Optional
+
+import yaml
+import json
+import dataclasses
+from krkn_lib.models.telemetry import ChaosRunTelemetry
+
+from krkn_lib.prometheus.krkn_prometheus import KrknPrometheus
+from krkn.prometheus.collector import evaluate_slos
+from krkn.resiliency.score import calculate_resiliency_score
+
+
+class Resiliency:  
+    """Central orchestrator for resiliency scoring."""
+
+
+
+
+    ENV_VAR_NAME = "KRKN_ALERTS_YAML_CONTENT"
+
+    def __init__(self, alerts_yaml_path: str = "config/alerts.yaml"):
+        """Load SLO definitions from the default alerts file, unless the
+        *KRKN_ALERTS_YAML_CONTENT* environment variable is set – in which case its
+        raw YAML string is parsed instead. The custom YAML may optionally follow
+        this schema:
+
+        prometheus_url: http://prometheus:9090  # optional, currently unused
+        slos:
+          - expr: <PromQL>
+            severity: critical|warning
+            description: <text>
+
+        For backward-compatibility the legacy list-only format is still accepted.
+        """
+        raw_yaml_data: Any
+        env_yaml = os.getenv(self.ENV_VAR_NAME)
+        if env_yaml:
+            try:
+                decoded_yaml = base64.b64decode(env_yaml).decode('utf-8')
+                raw_yaml_data = yaml.safe_load(decoded_yaml)
+            except (yaml.YAMLError, base64.binascii.Error, UnicodeDecodeError) as exc:
+                raise ValueError(
+                    f"Invalid base64 or YAML in environment variable {self.ENV_VAR_NAME}: {exc}"
+                ) from exc
+            logging.info("Loaded SLO configuration from environment variable %s", self.ENV_VAR_NAME)
+            # Store optional Prometheus URL if provided
+            if isinstance(raw_yaml_data, dict):
+                self.prometheus_url = raw_yaml_data.get("prometheus_url")  # may be None
```

Comment:
the prometheus url should always come from the config/config.yaml file, don't think this will be used